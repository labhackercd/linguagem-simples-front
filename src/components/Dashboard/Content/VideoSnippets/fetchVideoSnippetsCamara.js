import axiosInstance from '../../../../auth/axiosApi'
import {API_VIDEOS_SNIPPETS} from '../../../../api_urls'


export default async function fetchSessionVideos(sessionIdDadosAbertos){
    const response = await axiosInstance.get((API_VIDEOS_SNIPPETS+sessionIdDadosAbertos), {});

    if(response.status===200){
        return response.data;
    }else{
        return null;
    }
}
