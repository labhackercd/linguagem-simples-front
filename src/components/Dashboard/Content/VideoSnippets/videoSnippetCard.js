import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import ButtonBase from '@material-ui/core/ButtonBase'
import VideoSnippetModal from './VideoSnippetModal/videoSnippetModal'

export default function SnippetCard(props){
  const deputado = props.data.author;
  const siglaPartidaria = props.data.legend;

  const [open, setOpen] = React.useState(false);


  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = (value) => {
    setOpen(false);
  };

  return (
    <Box width="100%" height="100%" marginX={0.3} marginTop={0.5}>
      <ButtonBase onClick={handleClickOpen} id={"openModal"}>
        <div>
            <img style={{width:"100%", height:"6vh"}} id="image" src={props.data.thumbnail} alt="Thumbanail trecho"></img>
            <Box fontSize={10}>
            <Typography>{deputado}</Typography>
            </Box>
            <Box fontSize={8}>
            <Typography variant="body2">{siglaPartidaria}</Typography>
            </Box>
        </div>
        </ButtonBase>
      <VideoSnippetModal open={open} onClose={handleClose} data={props.data} />
    </Box>
  );
}