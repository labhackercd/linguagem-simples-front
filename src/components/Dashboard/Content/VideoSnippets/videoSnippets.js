import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import CircularProgress from '@material-ui/core/CircularProgress';
import {videoSnippetsMockData} from './videoSnippetsMockData'
import fetchSessionVideos from './fetchVideoSnippetsCamara'
import { FixedSizeList } from 'react-window';
import ListItem from '@material-ui/core/ListItem';
import InputAdornment from '@material-ui/core/InputAdornment';
import SearchIcon from '@material-ui/icons/Search';
import TextField from '@material-ui/core/TextField';
import ButtonBase from '@material-ui/core/ButtonBase'

import VideoSnippetModal from './VideoSnippetModal/videoSnippetModal'
import SnippetCard from './videoSnippetCard'



export default class VideoSnippets extends React.Component {
    
  constructor(props){
    super(props);
    this.state = { 
        snippets: '',
        filteredSnippets: '',
        sessionHasStarted: false,
        dataLoaded: false,
        searchField: '',
        error:{
          happened:false,
          message: "Erro",
          status:""
        }
    };
  }

  componentDidMount(){
      this._isMounted = true;
      //console.log(this.props)
      if(this._isMounted){

          if(this.props.sessionInfo.situation_session === "pre_session"){
            this.setState({sessionHasStarted:false});
            //this.setState({snippets:true});
          }else{
            this.setState({sessionHasStarted:true})
            this.fetchSessionsList();
          }
          
      }
  }

  fetchSessionsList = async term => {
    try {
      //console.log(this.props)
      const data = await fetchSessionVideos(this.props.sessionInfo.id_session_dados_abertos);
      this.setState({snippets:data})
      this.setState({filteredSnippets:data})
      this.setState({dataLoaded:true});
    } catch (error) {
      //console.log(error)
      this.setState({error:{happened:true,status:error.response.status} })
    }
  };

  renderSearchBarFunction(){
    return(
      <React.Fragment>
        <Grid container>
          <Grid item xs={10}>
            <Box marginLeft={2} marginTop={1}>
              <Typography variant="h5" style={{ color: "#3E3E3E" }}>Trechos</Typography>
            </Box>
          </Grid>
          <Grid item xs={2}>
            <Box marginRight={1}> 
              <TextField
                id="input-search-snippet"
                size="small"
                onChange={this.snippetsFilterOnChange}
                InputProps={{
                  endAdornment: (<InputAdornment position="start"><SearchIcon /></InputAdornment>),
                }}/>
            </Box>
          </Grid>
        </Grid>
      </React.Fragment>
    )
  }

  renderListItem = ({index, style}) => {
    return(
      <ListItem disableGutters={true} style={style} >
          <SnippetCard data = {this.state.filteredSnippets[index]}></SnippetCard>        
      </ListItem>
    )
  }

  snippetsFilterOnChange = (event) => {
    //console.log(event.target.value)
    this.setState({
      searchField: event.target.value
    })
    
    this.setState({
      filteredSnippets: this.state.snippets.filter(term => term.author.toLowerCase().includes(this.state.searchField.toLowerCase()))
    })
    
  }

  render(){
    const widthSnippetsBox = parseInt(((window.innerWidth)*0.54));
    const widthSnippetsItem = (((window.innerWidth)*0.5)*0.12);
    //console.log(widthSnippetsBox)
   
    if(this.state.error.happened){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}>
                  <Typography>Erro</Typography>
              </Box>)
    }

    if(!this.state.sessionHasStarted){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}>
                  <Typography>A transmissão ainda não foi iniciada.</Typography>
              </Box>)
    }

    if(!this.state.dataLoaded){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}><CircularProgress></CircularProgress></Box>)
    }

    return (
      <div>
        <Grid container >
        {this.renderSearchBarFunction()}
          <Grid item xs={12}>
            <Box marginLeft={1}>
              <FixedSizeList width={widthSnippetsBox} height={"12.5vh"} itemSize={widthSnippetsItem} layout="horizontal" itemCount={this.state.filteredSnippets.length}>        
                {this.renderListItem}
              </FixedSizeList>
            </Box>
          </Grid>
        </Grid>
      </div>
    )
  }
}