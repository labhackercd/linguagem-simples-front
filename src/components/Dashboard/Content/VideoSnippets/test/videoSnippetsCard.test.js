import React from 'react';
import { render, screen } from '@testing-library/react';
import {shallow} from "enzyme/build";
import VideoSnippets from '../videoSnippets'
import { mount} from 'enzyme';
import MockAdapter from "axios-mock-adapter"
import CircularProgress from '@material-ui/core/CircularProgress';
import SnippetCard from '../videoSnippetCard'
import {API_VIDEOS_SNIPPETS} from '../../../../../api_urls'
import axiosInstance from '../../../../../auth/axiosApi'

  

describe('Testing VideoSnippetCard Component', () => {

    const mockedData={
        "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234640207",
        "author": "Arthur Oliveira Maia",
        "legend": "Deputado DEM-BA",
        "schedule": "horário - 15h57'20",
        "duration": "duração - 00:02:04",
        "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160600.jpg"
    }

    it("should render the Card component and match snapshot ", () => {
        
        const component = mount(<SnippetCard data={mockedData}/>);

        expect(component.exists()).toEqual(true);
        expect(component).toMatchSnapshot();
        
    });
    
    test("Test Card handleOpen and handleClose", () => {
        
        
        const component =  mount(<SnippetCard data={mockedData}/>);

        //console.log(component.debug())
        const button = component.find("#openModal").at(0);
        //console.log(button.debug())
        button.simulate('click')
        component.update();
        //console.log(component.debug())
        const closeButton = component.find("#closeVideoSnippetModal").at(0)
        closeButton.simulate('click')

    });

  });
  