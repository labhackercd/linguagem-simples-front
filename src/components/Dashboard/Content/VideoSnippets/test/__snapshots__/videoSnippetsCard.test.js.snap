// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoSnippetCard Component should render the Card component and match snapshot  1`] = `
<SnippetCard
  data={
    Object {
      "author": "Arthur Oliveira Maia",
      "duration": "duração - 00:02:04",
      "legend": "Deputado DEM-BA",
      "schedule": "horário - 15h57'20",
      "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160600.jpg",
      "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234640207",
    }
  }
>
  <Styled(MuiBox)
    height="100%"
    marginTop={0.5}
    marginX={0.3}
    width="100%"
  >
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <WithStyles(ForwardRef(ButtonBase))
        id="openModal"
        onClick={[Function]}
      >
        <ForwardRef(ButtonBase)
          classes={
            Object {
              "disabled": "Mui-disabled",
              "focusVisible": "Mui-focusVisible",
              "root": "MuiButtonBase-root",
            }
          }
          id="openModal"
          onClick={[Function]}
        >
          <button
            className="MuiButtonBase-root"
            disabled={false}
            id="openModal"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div>
              <img
                alt="Thumbanail trecho"
                id="image"
                src="https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160600.jpg"
                style={
                  Object {
                    "height": "6vh",
                    "width": "100%",
                  }
                }
              />
              <Styled(MuiBox)
                fontSize={10}
              >
                <div
                  className="MuiBox-root MuiBox-root-2"
                >
                  <WithStyles(ForwardRef(Typography))
                    variant="body2"
                  >
                    <ForwardRef(Typography)
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter",
                          "alignJustify": "MuiTypography-alignJustify",
                          "alignLeft": "MuiTypography-alignLeft",
                          "alignRight": "MuiTypography-alignRight",
                          "body1": "MuiTypography-body1",
                          "body2": "MuiTypography-body2",
                          "button": "MuiTypography-button",
                          "caption": "MuiTypography-caption",
                          "colorError": "MuiTypography-colorError",
                          "colorInherit": "MuiTypography-colorInherit",
                          "colorPrimary": "MuiTypography-colorPrimary",
                          "colorSecondary": "MuiTypography-colorSecondary",
                          "colorTextPrimary": "MuiTypography-colorTextPrimary",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary",
                          "displayBlock": "MuiTypography-displayBlock",
                          "displayInline": "MuiTypography-displayInline",
                          "gutterBottom": "MuiTypography-gutterBottom",
                          "h1": "MuiTypography-h1",
                          "h2": "MuiTypography-h2",
                          "h3": "MuiTypography-h3",
                          "h4": "MuiTypography-h4",
                          "h5": "MuiTypography-h5",
                          "h6": "MuiTypography-h6",
                          "noWrap": "MuiTypography-noWrap",
                          "overline": "MuiTypography-overline",
                          "paragraph": "MuiTypography-paragraph",
                          "root": "MuiTypography-root",
                          "srOnly": "MuiTypography-srOnly",
                          "subtitle1": "MuiTypography-subtitle1",
                          "subtitle2": "MuiTypography-subtitle2",
                        }
                      }
                      variant="body2"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2"
                      >
                        Arthur Oliveira Maia
                      </p>
                    </ForwardRef(Typography)>
                  </WithStyles(ForwardRef(Typography))>
                </div>
              </Styled(MuiBox)>
              <Styled(MuiBox)
                fontSize={8}
              >
                <div
                  className="MuiBox-root MuiBox-root-3"
                >
                  <WithStyles(ForwardRef(Typography))
                    variant="body2"
                  >
                    <ForwardRef(Typography)
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter",
                          "alignJustify": "MuiTypography-alignJustify",
                          "alignLeft": "MuiTypography-alignLeft",
                          "alignRight": "MuiTypography-alignRight",
                          "body1": "MuiTypography-body1",
                          "body2": "MuiTypography-body2",
                          "button": "MuiTypography-button",
                          "caption": "MuiTypography-caption",
                          "colorError": "MuiTypography-colorError",
                          "colorInherit": "MuiTypography-colorInherit",
                          "colorPrimary": "MuiTypography-colorPrimary",
                          "colorSecondary": "MuiTypography-colorSecondary",
                          "colorTextPrimary": "MuiTypography-colorTextPrimary",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary",
                          "displayBlock": "MuiTypography-displayBlock",
                          "displayInline": "MuiTypography-displayInline",
                          "gutterBottom": "MuiTypography-gutterBottom",
                          "h1": "MuiTypography-h1",
                          "h2": "MuiTypography-h2",
                          "h3": "MuiTypography-h3",
                          "h4": "MuiTypography-h4",
                          "h5": "MuiTypography-h5",
                          "h6": "MuiTypography-h6",
                          "noWrap": "MuiTypography-noWrap",
                          "overline": "MuiTypography-overline",
                          "paragraph": "MuiTypography-paragraph",
                          "root": "MuiTypography-root",
                          "srOnly": "MuiTypography-srOnly",
                          "subtitle1": "MuiTypography-subtitle1",
                          "subtitle2": "MuiTypography-subtitle2",
                        }
                      }
                      variant="body2"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2"
                      >
                        15h57'20
                      </p>
                    </ForwardRef(Typography)>
                  </WithStyles(ForwardRef(Typography))>
                </div>
              </Styled(MuiBox)>
            </div>
            <WithStyles(memo)
              center={false}
            >
              <ForwardRef(TouchRipple)
                center={false}
                classes={
                  Object {
                    "child": "MuiTouchRipple-child",
                    "childLeaving": "MuiTouchRipple-childLeaving",
                    "childPulsate": "MuiTouchRipple-childPulsate",
                    "ripple": "MuiTouchRipple-ripple",
                    "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                    "rippleVisible": "MuiTouchRipple-rippleVisible",
                    "root": "MuiTouchRipple-root",
                  }
                }
              >
                <span
                  className="MuiTouchRipple-root"
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    component={null}
                    exit={true}
                  />
                </span>
              </ForwardRef(TouchRipple)>
            </WithStyles(memo)>
          </button>
        </ForwardRef(ButtonBase)>
      </WithStyles(ForwardRef(ButtonBase))>
      <VideoSnippetModal
        data={
          Object {
            "author": "Arthur Oliveira Maia",
            "duration": "duração - 00:02:04",
            "legend": "Deputado DEM-BA",
            "schedule": "horário - 15h57'20",
            "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160600.jpg",
            "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234640207",
          }
        }
        onClose={[Function]}
        open={false}
      >
        <WithStyles(ForwardRef(Dialog))
          aria-labelledby="simple-dialog-title"
          fullWidth={true}
          maxWidth="md"
          onClose={[Function]}
          open={false}
        >
          <ForwardRef(Dialog)
            aria-labelledby="simple-dialog-title"
            classes={
              Object {
                "container": "MuiDialog-container",
                "paper": "MuiDialog-paper",
                "paperFullScreen": "MuiDialog-paperFullScreen",
                "paperFullWidth": "MuiDialog-paperFullWidth",
                "paperScrollBody": "MuiDialog-paperScrollBody",
                "paperScrollPaper": "MuiDialog-paperScrollPaper",
                "paperWidthFalse": "MuiDialog-paperWidthFalse",
                "paperWidthLg": "MuiDialog-paperWidthLg",
                "paperWidthMd": "MuiDialog-paperWidthMd",
                "paperWidthSm": "MuiDialog-paperWidthSm",
                "paperWidthXl": "MuiDialog-paperWidthXl",
                "paperWidthXs": "MuiDialog-paperWidthXs",
                "root": "MuiDialog-root",
                "scrollBody": "MuiDialog-scrollBody",
                "scrollPaper": "MuiDialog-scrollPaper",
              }
            }
            fullWidth={true}
            maxWidth="md"
            onClose={[Function]}
            open={false}
          >
            <ForwardRef(Modal)
              BackdropComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "Naked": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "propTypes": Object {
                      "children": [Function],
                      "className": [Function],
                      "classes": [Function],
                      "invisible": [Function],
                      "open": [Function],
                      "transitionDuration": [Function],
                    },
                    "render": [Function],
                  },
                  "displayName": "WithStyles(ForwardRef(Backdrop))",
                  "options": Object {
                    "defaultTheme": Object {
                      "breakpoints": Object {
                        "between": [Function],
                        "down": [Function],
                        "keys": Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ],
                        "only": [Function],
                        "up": [Function],
                        "values": Object {
                          "lg": 1280,
                          "md": 960,
                          "sm": 600,
                          "xl": 1920,
                          "xs": 0,
                        },
                        "width": [Function],
                      },
                      "direction": "ltr",
                      "mixins": Object {
                        "gutters": [Function],
                        "toolbar": Object {
                          "@media (min-width:0px) and (orientation: landscape)": Object {
                            "minHeight": 48,
                          },
                          "@media (min-width:600px)": Object {
                            "minHeight": 64,
                          },
                          "minHeight": 56,
                        },
                      },
                      "overrides": Object {},
                      "palette": Object {
                        "action": Object {
                          "activatedOpacity": 0.12,
                          "active": "rgba(0, 0, 0, 0.54)",
                          "disabled": "rgba(0, 0, 0, 0.26)",
                          "disabledBackground": "rgba(0, 0, 0, 0.12)",
                          "disabledOpacity": 0.38,
                          "focus": "rgba(0, 0, 0, 0.12)",
                          "focusOpacity": 0.12,
                          "hover": "rgba(0, 0, 0, 0.04)",
                          "hoverOpacity": 0.04,
                          "selected": "rgba(0, 0, 0, 0.08)",
                          "selectedOpacity": 0.08,
                        },
                        "augmentColor": [Function],
                        "background": Object {
                          "default": "#fafafa",
                          "paper": "#fff",
                        },
                        "common": Object {
                          "black": "#000",
                          "white": "#fff",
                        },
                        "contrastThreshold": 3,
                        "divider": "rgba(0, 0, 0, 0.12)",
                        "error": Object {
                          "contrastText": "#fff",
                          "dark": "#d32f2f",
                          "light": "#e57373",
                          "main": "#f44336",
                        },
                        "getContrastText": [Function],
                        "grey": Object {
                          "100": "#f5f5f5",
                          "200": "#eeeeee",
                          "300": "#e0e0e0",
                          "400": "#bdbdbd",
                          "50": "#fafafa",
                          "500": "#9e9e9e",
                          "600": "#757575",
                          "700": "#616161",
                          "800": "#424242",
                          "900": "#212121",
                          "A100": "#d5d5d5",
                          "A200": "#aaaaaa",
                          "A400": "#303030",
                          "A700": "#616161",
                        },
                        "info": Object {
                          "contrastText": "#fff",
                          "dark": "#1976d2",
                          "light": "#64b5f6",
                          "main": "#2196f3",
                        },
                        "primary": Object {
                          "contrastText": "#fff",
                          "dark": "#303f9f",
                          "light": "#7986cb",
                          "main": "#3f51b5",
                        },
                        "secondary": Object {
                          "contrastText": "#fff",
                          "dark": "#c51162",
                          "light": "#ff4081",
                          "main": "#f50057",
                        },
                        "success": Object {
                          "contrastText": "rgba(0, 0, 0, 0.87)",
                          "dark": "#388e3c",
                          "light": "#81c784",
                          "main": "#4caf50",
                        },
                        "text": Object {
                          "disabled": "rgba(0, 0, 0, 0.38)",
                          "hint": "rgba(0, 0, 0, 0.38)",
                          "primary": "rgba(0, 0, 0, 0.87)",
                          "secondary": "rgba(0, 0, 0, 0.54)",
                        },
                        "tonalOffset": 0.2,
                        "type": "light",
                        "warning": Object {
                          "contrastText": "rgba(0, 0, 0, 0.87)",
                          "dark": "#f57c00",
                          "light": "#ffb74d",
                          "main": "#ff9800",
                        },
                      },
                      "props": Object {},
                      "shadows": Array [
                        "none",
                        "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
                        "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
                        "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
                        "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
                        "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
                        "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
                        "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
                        "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
                        "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
                        "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
                        "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
                        "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
                        "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
                        "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
                        "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
                        "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
                        "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
                        "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
                        "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
                        "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
                        "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
                        "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
                        "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
                        "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
                      ],
                      "shape": Object {
                        "borderRadius": 4,
                      },
                      "spacing": [Function],
                      "transitions": Object {
                        "create": [Function],
                        "duration": Object {
                          "complex": 375,
                          "enteringScreen": 225,
                          "leavingScreen": 195,
                          "short": 250,
                          "shorter": 200,
                          "shortest": 150,
                          "standard": 300,
                        },
                        "easing": Object {
                          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
                          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
                          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
                          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
                        },
                        "getAutoHeightDuration": [Function],
                      },
                      "typography": Object {
                        "body1": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "1rem",
                          "fontWeight": 400,
                          "letterSpacing": "0.00938em",
                          "lineHeight": 1.5,
                        },
                        "body2": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "0.875rem",
                          "fontWeight": 400,
                          "letterSpacing": "0.01071em",
                          "lineHeight": 1.43,
                        },
                        "button": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "0.875rem",
                          "fontWeight": 500,
                          "letterSpacing": "0.02857em",
                          "lineHeight": 1.75,
                          "textTransform": "uppercase",
                        },
                        "caption": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "0.75rem",
                          "fontWeight": 400,
                          "letterSpacing": "0.03333em",
                          "lineHeight": 1.66,
                        },
                        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                        "fontSize": 14,
                        "fontWeightBold": 700,
                        "fontWeightLight": 300,
                        "fontWeightMedium": 500,
                        "fontWeightRegular": 400,
                        "h1": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "6rem",
                          "fontWeight": 300,
                          "letterSpacing": "-0.01562em",
                          "lineHeight": 1.167,
                        },
                        "h2": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "3.75rem",
                          "fontWeight": 300,
                          "letterSpacing": "-0.00833em",
                          "lineHeight": 1.2,
                        },
                        "h3": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "3rem",
                          "fontWeight": 400,
                          "letterSpacing": "0em",
                          "lineHeight": 1.167,
                        },
                        "h4": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "2.125rem",
                          "fontWeight": 400,
                          "letterSpacing": "0.00735em",
                          "lineHeight": 1.235,
                        },
                        "h5": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "1.5rem",
                          "fontWeight": 400,
                          "letterSpacing": "0em",
                          "lineHeight": 1.334,
                        },
                        "h6": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "1.25rem",
                          "fontWeight": 500,
                          "letterSpacing": "0.0075em",
                          "lineHeight": 1.6,
                        },
                        "htmlFontSize": 16,
                        "overline": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "0.75rem",
                          "fontWeight": 400,
                          "letterSpacing": "0.08333em",
                          "lineHeight": 2.66,
                          "textTransform": "uppercase",
                        },
                        "pxToRem": [Function],
                        "round": [Function],
                        "subtitle1": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "1rem",
                          "fontWeight": 400,
                          "letterSpacing": "0.00938em",
                          "lineHeight": 1.75,
                        },
                        "subtitle2": Object {
                          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                          "fontSize": "0.875rem",
                          "fontWeight": 500,
                          "letterSpacing": "0.00714em",
                          "lineHeight": 1.57,
                        },
                      },
                      "zIndex": Object {
                        "appBar": 1100,
                        "drawer": 1200,
                        "mobileStepper": 1000,
                        "modal": 1300,
                        "snackbar": 1400,
                        "speedDial": 1050,
                        "tooltip": 1500,
                      },
                    },
                    "name": "MuiBackdrop",
                  },
                  "propTypes": Object {
                    "classes": [Function],
                    "innerRef": [Function],
                  },
                  "render": [Function],
                  "useStyles": [Function],
                }
              }
              BackdropProps={
                Object {
                  "transitionDuration": Object {
                    "enter": 225,
                    "exit": 195,
                  },
                }
              }
              className="MuiDialog-root"
              closeAfterTransition={true}
              disableBackdropClick={false}
              disableEscapeKeyDown={false}
              onClose={[Function]}
              open={false}
            />
          </ForwardRef(Dialog)>
        </WithStyles(ForwardRef(Dialog))>
      </VideoSnippetModal>
    </div>
  </Styled(MuiBox)>
</SnippetCard>
`;
